### DISCUSSION TITLE
S100103  - Bestway Ibiza - Works with direct connections; without PCB

### DISCUSSION THREAD

**Discussion by deeja** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#383] [Show and tell]
![image](https://user-images.githubusercontent.com/174247/193428536-d0cd273b-18d5-4ac8-9eeb-a81b21a68823.png)

Since the IO pins of the NodeMCU are [5v tolerant](https://www.qworqs.com/2021/05/19/are-the-esp32-and-esp8266-5v-tolerant-yes-they-officially-are/) I thought I would connect up directly. 

![image](https://user-images.githubusercontent.com/174247/193428642-b86009a6-062b-44da-a768-b127db6f213c.png)

Set the model to MIAMI2021 in `Model.h` (I didn't read the instructions and this cost me a lot of time)
```cpp
#define MIAMI2021  //no hydrojets
```

Used my own pin IO setup
```cpp
  // D1 = cio_cs_pin
  // D2 = cio_clk_pin
  // D3 = cio_data_pin  
  // D5 = dsp_data_pin 
  // D6 = dsp_clk_pin 
  // D7 = dsp_cs_pin 
  // D8 = dsp_audio_pin
  bwc.begin(D3,D2,D1,D5,D6,D7,D8 ); 
```

![image](https://user-images.githubusercontent.com/174247/193429135-607cf7dd-7d77-4020-9688-4d6cd8b6c603.png)

**Comment by rosetown-kelly** [comment]
Very cool. Any downside to doing it this way? 

Many have mentioned here that not using a level shifter would result in a fried board, so I'm a bit surprised to see this!

**Reply by rosetown-kelly** [reply]
@visualapproach curious your thoughts on this. I also just wired my model S100104 the same way as @deeja and it does appear to work just fine.

**Reply by visualapproach** [reply]
@rosetown-kelly the 8266 is a 3.3V  device which apparently survives 5V on the inputs. Unless espressif (the maker) guarantees this I wouldn't trust it for a pacemaker but for this application you're only risking a $5 part and you save half by not using an LLC.
The pump is a 5V system but luckily it reads the 3.3V as a logic high.

**Reply by deeja** [reply]
CEO of Expressif confirmed 5v tolerance on the espressif chips a few years ago now. https://www.facebook.com/groups/1499045113679103/permalink/1731855033731442/
![image](https://user-images.githubusercontent.com/174247/193750069-6e95e0f8-9ff3-4deb-bd6a-eb2682c3fe55.png)

**Reply by deeja** [reply]
@visualapproach  - really enjoying the control over the wifi! Thanks for making this happen

**Reply by deeja** [reply]
> Very cool. Any downside to doing it this way?

Since the voltage can drop over a distance, a possible downside to outputting 3.3v high is the that it may drop over a distance and might not trigger the pump's high state. I'm going to do a test run at some point to see how far I can extend the controls from the pump using this setup via a few metre long 6 core cable

**Reply by visualapproach** [reply]
thanks for the info, and let us know how your long range test goes :-)

**Reply by deeja** [reply]
![image](https://user-images.githubusercontent.com/174247/195215489-4af5492f-a06a-49ec-ae71-dda9972217a6.png)

Long range test is so/so at the moment. I've extended out the controls by 2m using 6 core shielded computer cable.

Good news is that all the commands, including the scheduled commands, are working as they should. 
Not so great news is that the wifi is acting a bit strange. I'm not sure if it is the esp, the enclosure or my router that is the issue. Possibly lack of power to the chip? 

I initially stuffed the nodemcu into the control panel, and it did work but the wifi connection was very poor. 
An external antenna model would likely have better results.

**Reply by deeja** [reply]
![image](https://user-images.githubusercontent.com/174247/195216145-af284f4e-9b2e-469e-8744-11d085ac84e1.png)
Maybe too much stuff in there
